### PR TITLE
piv: block use of RSA1024 keys on YubiKey FIPS

### PR DESCRIPTION
**Pull request by dagheyman**

### CODE REVIEW

**Comment by dagheyman** [comment]
(not ready yet)

File: ykman/cli/piv.py [hunk 1]
```diff
@@ -792,10 +794,17 @@ def _authenticate(ctx, controller, management_key, mgm_key_prompt,
...
     #  ECCP384 not supported on NEO.
     if algorithm == ALGO.ECCP384 and controller.version < (4, 0, 0):
         ctx.fail('ECCP384 is not supported by this YubiKey.')
+    if algorithm == ALGO.RSA1024 and ctx.obj['dev'].is_fips:
+        ctx.fail('RSA1024 is not supoported by this YubiKey.')
+
+
+def _check_key_size(ctx, private_key):
+    if ctx.obj['dev'].is_fips and private_key.key_size == 1024:
+        ctx.fail('1024 is not a supoported key size on this YubiKey.')
```
Comment (emlun) [inline ykman/cli/piv.py]: Typo: supoported -> supported

File: ykman/cli/piv.py [hunk 2]
```diff
@@ -792,10 +794,17 @@ def _authenticate(ctx, controller, management_key, mgm_key_prompt,
         ctx.fail('Authentication with management key failed.')
 
 
-def _check_eccp384(ctx, controller, algorithm):
+def _check_algorithm(ctx, controller, algorithm):
     #  ECCP384 not supported on NEO.
     if algorithm == ALGO.ECCP384 and controller.version < (4, 0, 0):
         ctx.fail('ECCP384 is not supported by this YubiKey.')
+    if algorithm == ALGO.RSA1024 and ctx.obj['dev'].is_fips:
+        ctx.fail('RSA1024 is not supoported by this YubiKey.')
```
Comment (emlun) [inline ykman/cli/piv.py]: Typo: supoported -> supported

**Comment by dagheyman** [comment]
Added a couple of tests and fixed the typo.

> On second thought, shouldn't this be done in the library instead of the CLI?

Maybe, but it fitted well next to the other checks in `cli/piv.py`. Let's discuss that more and maybe do separate refactoring PRs to move these kind of checks to the library level, if that is what we want.

**Comment by emlun** [comment]
Sounds good.

**Pull request merged**